app/services/messaging/channel.rb

```diff
@@ -9,31 +9,38 @@ def initialize
     @metrics = Metrics.with_object(self)
   end
 
+  delegate :transaction, to: ActiveRecord::Base
```

Comment:
Seems a bit overkill to set up this delegation for one usage. Let's wait until there's more. And even then, should we set up a `with_transaction` method or something, instead of delegation? Delegation tends to make tracing codepaths harder through a codebase, I think.